receipt_layoutlm/receipt_layoutlm/export_coreml.py

```
@@ -226,6 +230,25 @@ def export_coreml(
+    )
+    if weight_path.exists():
+        raw = weight_path.read_bytes()
+        fp16_arr = np.frombuffer(raw, dtype=np.float16)
+        nan_count = int(np.isnan(fp16_arr).sum())
+        if nan_count > 0:
+            raise NaNWeightsError(
+                f"Exported CoreML model contains {nan_count} NaN values in "
+                f"weight.bin ({weight_path}). This is typically caused by "
+                f"float16 quantization overflowing large weight values. "
+                f"Re-export without --quantize float16."
+            )
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)